pkg/apply/task/inv_add_task.go

```diff
@@ -71,7 +72,16 @@ func (i *InvAddTask) Start(taskContext *taskrunner.TaskContext) {
 		}
 		klog.V(4).Infof("merging %d local objects into inventory", len(i.Objects))
 		currentObjs := object.UnstructuredSetToObjMetadataSet(i.Objects)
-		_, err := i.InvClient.Merge(taskContext.Context(), i.InvInfo, currentObjs, i.DryRun)
+		inventoryObjs := i.ClusterInventory.Objects()
+		unionObjs := inventoryObjs.Union(currentObjs)
+
+		i.ClusterInventory.SetObjects(unionObjs)
+
+		var err error
+		if !i.DryRun.ClientOrServerDryRun() {
+			err = i.InvClient.CreateOrUpdate(taskContext.Context(), i.ClusterInventory, inventory.UpdateOptions{})
```

Comment:
I think we talked about this, but what happens if the inventory has changed between the Get at the beginning of the Apply/Destroy and this Update call? Conflict, error, and exit?

How many other API calls are happening between these two calls?

Example cases we could test for: 
- a controller (ex: ResourceGroup controller) updates the status
- a controller updates the metadata
- a mutating webhook adds metadata to the inventory object

Reply:
Has this been solved? Not sure where to expect the retry. Here? Or in CreateOrUpdate?

Reply:
Yes, there is a retry in CreateOrUpdate.